BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -274,7 +250,7 @@ - (void)failAuthenticationWithConnection:(NSURLConnection *)connection
             self.completionBlock(self, nil, self.connectionError);
         }
     }
-    
+
```

Comment:
should we call the self.completionBlock only when sessionTask finishes in the callback - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error?

if given our contract defines completionBlock as the very last step to be called processing user's request. We don't know what might be a side effect of executing completionBlock's logic to the remaining life cycle of session task.